src/main/java/uk/ac/ebi/ena/readtools/loader/common/producer/ProducibleData.java

```diff
@@ -8,17 +8,24 @@
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 */
-package uk.ac.ebi.ena.readtools.loader.common.feeder;
+package uk.ac.ebi.ena.readtools.loader.common.producer;
 
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * A member field of a class annotated with this is expected to hold data that will be acquired by calling the method
+ * whose name is defined in the property {@link ProducibleData#method}.
+ */
 @Retention( RetentionPolicy.RUNTIME )
 @Target( {ElementType.FIELD } )
 public @interface
-FeedableData
+ProducibleData

```

Comment:
Is it possible to replace ProducibleData annotation with a an interface with the required methods?